src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateArray = [];
+
+  let stateCopy = {
+    ...state,
+  };
+
+  for (const part of actions) {
+    switch (part.type) {
+      case 'addProperties':
+        stateCopy = {
+          ...stateCopy,
+          ...part.extraData,
+        };
```

Comment:
You can use [Object.assign](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign) for that